laspy/header.py

```diff
@@ -743,6 +744,23 @@ def write_to(
         stream.write(vlr_bytes)
         stream.write(self.extra_vlr_bytes)
 
+    def parse_crs(self) -> Optional[pyproj.CRS]:
+        """Method to parse OGC WKT or GeoTiff VLR keys into a pyproj CRS object"""
+
+        geo_vlr = self._vlrs.get_by_id('LASF_Projection')
+
+        for rec in geo_vlr:
+            # perfer OGC WKT over GeoTiff
+            if type(rec) == WktCoordinateSystemVlr:
+                return pyproj.CRS.from_wkt(geo_vlr[0].string)
+            elif type(rec) == GeoKeyDirectoryVlr:
+                for key in rec.geo_keys:
+                    # get ProjectedCRSGeoKey by id
+                    if key.id == 3072:
+                        return pyproj.CRS.from_epsg(key.value_offset)
```

Comment:
What I would have done is create a `parse_crs` method in `WktCoordinateSystemVlr` :
```Python
class WktCoordinateSystemVlr:
# ...
def parse_crs(self) -> pyproj.CRS:
    return pyproj.CRS.from_wkt(self.string)
```

and one in `GeoKeyDirectoryVlr`
```Python
class GeoKeyDirectoryVlr:
# ...
def parse_crs(self) -> Optional[pyproj.CRS]:
      for key in self.geo_keys:
                    # get ProjectedCRSGeoKey by id
                    if key.id == 3072:
                        return pyproj.CRS.from_epsg(key.value_offset)
       return None  
```

And `LasHeader.parse_crs` would forward the call to these.

Reply:
I thought about doing the same thing, but went with the lazier option just in case you didn't want this :laughing: I'll split this up into the appropriate classes.